arangod/IResearch/VelocyPackHelper.cpp

```diff
@@ -149,6 +209,39 @@ bool mergeSliceSkipOffsets(velocypack::Builder& builder,
   return true;
 }
 
+bool parseDirectionBool(arangodb::velocypack::Slice slice, bool& direction) {
+  if (slice.isBool()) {
+    // true - asc
+    // false - desc
+    direction = slice.getBool();
+    return true;
+  }
+
+  // unsupported value type
+  return false;
+}
+
+bool parseDirectionString(arangodb::velocypack::Slice slice, bool& direction) {
+  if (slice.isString()) {
+    std::string value = slice.copyString();
+    arangodb::basics::StringUtils::tolowerInPlace(value);
+
+    if (value == "asc") {
+      direction = true;
+      return true;
+    }
+
+    if (value == "desc") {
+      direction = false;
+      return true;
+    }
```

Comment:
Here we will save string copy but do conversion twice. Not sure if it would be faster. Anyway as here is not performance-critical code I'd prefer to keep simpler code.